services/api/src/resources/task/resolvers.ts

```diff
@@ -329,6 +331,71 @@ export const deleteTask: ResolverFn = async (
   return 'success';
 };
 
+export const cancelTask: ResolverFn = async (
+  root,
+  {
+    input: {
+      task: taskInput,
+    }
+  },
+  { sqlClientPool, hasPermission, userActivityLogger }
+) => {
+
+  const task = await Helpers(sqlClientPool, hasPermission).getTaskByTaskInput(taskInput);
+  const environment = await environmentHelpers(sqlClientPool).getEnvironmentById(task.environment);
+  const project = await projectHelpers(sqlClientPool).getProjectById(environment.project);
+
+  if (!task) {
+    return null;
+  }
+
+  const envPerm = await environmentHelpers(sqlClientPool).getEnvironmentById(
```

Comment:
we already pull the environment on line 345 aren't we? Is there a side effect here or something that requires this to be pulled again?

Reply:
nah, just duped code